### PR TITLE
fix: only accept blocks with a height greater than tip - N 

### PR DESCRIPTION
**Pull request by u2**

### CODE REVIEW

File: sync/src/relayer/compact_block_process.rs [hunk 1]
```diff
@@ -57,6 +57,24 @@ impl<'a, CS: ChainStore + 'static> CompactBlockProcess<'a, CS> {
...
         let block_hash = compact_block.header.hash().to_owned();
 
+        // Only accept blocks with a height greater than tip - N
+        // where N is the current epoch length
+        let (lowest_number, tip) = {
+            let cs = self.relayer.shared.lock_chain_state();
+            let epoch_length = cs.current_epoch_ext().length();
+            let tip = cs.tip_header().clone();
+
+            if tip.number() > epoch_length {
```
Comment (doitian) [inline sync/src/relayer/compact_block_process.rs]: It is simpler to use `saturating_sub`

File: sync/src/relayer/error.rs [hunk 2]
```diff
@@ -42,4 +42,6 @@ pub enum Ignored {
     AlreadyInFlight,
     #[fail(display = "Already stored")]
     AlreadyStored,
+    #[fail(display = "Block is too early")]
+    TooEarlyBlock,
```
Comment (doitian) [inline sync/src/relayer/error.rs]: I think `TooOldBlock` is more accurate. Too early can be interpreted that the block is in the future, and comes too early.

File: sync/src/relayer/compact_block_process.rs (hunk 1 above)
Comment (u2) [reply sync/src/relayer/compact_block_process.rs]: thanks.

File: sync/src/relayer/error.rs (hunk 2 above)
Comment (u2) [reply sync/src/relayer/error.rs]: fixed

**Pull request merged**